### PR TITLE
tests: port additional snap-confine regression tests

### PR DESCRIPTION
**Pull request by zyga**

### CODE REVIEW

File: tests/lib/snaps.sh [hunk 1]
```diff
@@ -8,3 +8,12 @@ install_local() {
     fi
     snap install --dangerous "$SNAP_FILE"
 }
+
+install_local_devmode() {
```
Comment (fgimenez) [inline tests/lib/snaps.sh]: Couldn't we extend `install_local` so we don't duplicate code?

File: tests/regression/lp-1599891/task.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+summary: Regression check for https://bugs.launchpad.net/snap-confine/+bug/1599891
+execute: |
+    snap_confine=/usr/lib/snapd/snap-confine
+    echo "Seeing that snap-confine is in $snap_confine"
+
+    echo "I also see a corresponding apparmor profile"
+    grep -F -q "$snap_confine (enforce)" "/sys/kernel/security/apparmor/profiles"
```
Comment (fgimenez) [inline tests/regression/lp-1599891/task.yaml]: Can't we use `MATCH` here?

File: tests/lib/snaps.sh (hunk 1 above)
Comment (zyga) [inline tests/lib/snaps.sh]: Yep, done

File: tests/regression/lp-1599891/task.yaml (hunk 2 above)
Comment (zyga) [inline tests/regression/lp-1599891/task.yaml]: Done

**Pull request merged**